src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -291,13 +304,10 @@ protected void AddCoins(TransactionBuildContext context)
                     UnspentOutputReference item = availableHashList[outPoint];
 
                     coins.Add(new Coin(item.Transaction.Id, (uint)item.Transaction.Index, item.Transaction.Amount, item.Transaction.ScriptPubKey));
-                    sum += item.Transaction.Amount;
                 }
             }
 
-            foreach (UnspentOutputReference item in context.UnspentOutputs
-                .OrderByDescending(a => a.Confirmations > 0)
-                .ThenByDescending(a => a.Transaction.Amount))
```

Comment:
Also here, please explain the :"bug" you found with regards to this change.

Reply:
to posterity:
beside this code doing a coin-preselection that we don't want because it's a TransactionBuilder responsibility (it's CoinSelection to be precise), the bug here was on the OrderByDescending

`.OrderByDescending(a => a.Confirmations > 0)`
Basically that linq lambda returns the expression that has to be sorted.
Usually in that lambda you specify the field you want to be sorted, like 
`.OrderByDescending(a => a.Confirmations)`
In the above example we sort by Confirmations in descending order

the problem is that instead we were sorting the result of the expression `a.Confirmations > 0` that was returning a boolean instead and the sorting were applied on the bool instead of the confirmation number.

To feel what I said in an interactive way, I've created this little script
https://dotnetfiddle.net/aHkfBx

Reply:
Ok thanks 👍